framework/web_app/src/pages/StockOverview.vue

```diff
@@ -1,96 +1,260 @@
+
+                <!-- TODO: IMPLEMENT -->
+                <q-select
+                    label="Stock Group"
+                    v-model="createStockItemForm.stock_group_id"
+                ></q-select>
+
+                <!-- TODO: IMPLEMENT -->
+                <q-select
+                    label="Stock Location"
+                    v-model="createStockItemForm.stock_location_id"
+                ></q-select>
+
+                <q-card-actions align="right">
+                    <q-btn
+                        type="submit"
+                        align="right"
+                        color="cyan"
+                        label="Save & Continue"
+                        size="lg"
+                    />
+                    <q-btn
+                        type="submit"
+                        align="right"
+                        color="green"
+                        label="Save & Close"
+                        size="lg"
+                        v-close-popup
+                    />
+                </q-card-actions>
+            </q-form>
+        </q-card>
+    </q-dialog>
+
+    <q-dialog v-model="shouldDisplayAddToShoppingCartModal">
+        <q-card style="width: 700px; max-width: 80vw">
+            <q-card-section class="row items-center">
+                <span class="q-ml-sm">
+                    Add {{ 'stock item' }} to a shopping list.
+                </span>
+            </q-card-section>
+
+            <q-card-actions align="right">
+                <q-btn
+                    color="primary"
+                    flat
+                    label="Cancel"
+                    v-close-popup
+                />
+                <q-btn
+                    color="primary"
+                    flat
+                    label="Add to list"
+                    v-close-popup
+                />
+            </q-card-actions>
+        </q-card>
+    </q-dialog>
 </template>
+
+<script lang="ts" setup>
+    import { StockLevel } from 'src/models/StockLevel';
+    import { CreateStockItemCommand } from 'src/services/api/StockItemApiService';
+    import { useStockItemStore } from 'src/stores/stockItemStore';
+    import { useStockLevelStore } from 'src/stores/stockLevelStore';
+    import { Ref, ref } from 'vue';
+
+    // region Common
+
+    const stockItemStore = useStockItemStore();
+    const stockLevelStore = useStockLevelStore();
+
+    const { createStockItemAsync, updateStockLevelAsync, stockItems } =
+        stockItemStore;
+    const { getStockLevelColour, stockLevels } = stockLevelStore;
+
+    // end region Common
+
+    // region Create Stock Item
+
+    const shouldDisplayCreateStockItemModal = ref(false);
+
+    const defaultCreateStockItemForm = {
+        days_until_stocktake_alert: 0,
+        name: '',
+        stock_group_id: null,
+        stock_level_id: stockLevels[0]?.stock_level_id,
+        stock_location_id: null
+    };
+
+    const createStockItemForm: Ref<CreateStockItemCommand> = ref({
```

Comment:
Have a look at ref vs reactive.
Seems to not make much of a difference for your use case but worth learning.

Reply:
https://stackoverflow.com/questions/61452458/ref-vs-reactive-in-vue-3

In this case because the objects value was never being reassigned we dont need the overhead of ref so i have changed it.